File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> arr) {
    int answer = 0;
    
    int len = arr.size();
    
    answer = arr[0];
    
    for(int i=1; i<len; i++) 
    {
        int tmp = arr[i];
        
        //최소 공배수 구하기
        int cases = min(answer, tmp);
        int num1 =answer;
        int num2 =tmp;
        
        for(int i=2; i<=cases; i++) 
        {
            while(num1 %i == 0 && num2 % i == 0) 
            {
                num1 /= i;
                num2 /= i;
            }
        }
        
        answer = num1 * tmp;
    }
    
    
    return answer;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    var answer = 0;
    
    const len = arr.length;
    
    answer = arr[0];
    
    for(let i =1; i<len ; i++ ) {
        const tmp = arr[i];
        
        //최소 공배수 구하기
        const cases = Math.min(answer, tmp);
        let num1 = answer;
        let num2 = tmp;
        
        for(let i=2; i<=cases; i++) {
            while(num1 %i ===0 && num2 %i ===0) {
                num1 /=i;
                num2 /=i;
            }
        }
        
        answer = num1*tmp;
        
    }
    
    
    return answer;
}